api/types/role.go

```diff
@@ -784,8 +807,43 @@ func (r *RoleV5) CheckAndSetDefaults() error {
 		if r.Spec.Allow.DatabaseLabels == nil {
 			r.Spec.Allow.DatabaseLabels = Labels{Wildcard: []string{Wildcard}}
 		}
+
+		if len(r.Spec.Allow.KubernetesResources) == 0 {
+			r.Spec.Allow.KubernetesResources = []KubernetesResource{
+				{
+					Kind:      KindKubePod,
+					Namespace: Wildcard,
+					Name:      Wildcard,
+				},
+			}
+		}
 	case V4, V5:
 		// Labels default to nil/empty for v4+ roles
+
+		// Allow unrestricted access to all pods.
+		if len(r.Spec.Allow.KubernetesResources) == 0 {
```

Comment:
If this is not empty, should we do the same `validateKubeResources` validation as in the V6 case below? Same above.